code/modules/awaymissions/mission_code/UO71-terrorspiders.dm

```diff
@@ -191,26 +191,33 @@
 	origin_tech = null
 	selfcharge = 1
 	can_charge = 0
-	var/inawaymission = 1
+	var/inawaymission = TRUE
+
+/obj/item/gun/energy/laser/awaymission_aeg/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, list(COMSIG_MOVABLE_Z_CHANGED), .proc/updateZ)
+	// Force update it incase it spawns outside an away mission and shouldnt be charged
+	updateZ()
 
 /obj/item/gun/energy/laser/awaymission_aeg/process()
+	// Only call the self-recharge proc if inside an away mission
+	if(inawaymission)
+		..()
+
+/obj/item/gun/energy/laser/awaymission_aeg/proc/updateZ()
```

Comment:
Now that I think about it.....why not just do 

`/obj/item/gun/energy/laser/awaymission_aeg/onTransitZ(old_z, new_z)`

and have all the code below run.

Then you're not having to register signals or send signals or anything.

Reply:
`onTransitZ` is now used to cut down on the overhead, even though its minimal.